pkg/server/service/loadbalancer/wrr/wrr.go

```diff
@@ -151,5 +149,15 @@ func (b *Balancer) AddService(name string, handler http.Handler, weight *int) {
 	if weight != nil {
 		w = *weight
 	}
-	b.handlers = append(b.handlers, &namedHandler{Handler: handler, name: name, weight: w})
+	if w <= 0 { // non-positive weight is meaningless
+		return
+	}
+	h := &namedHandler{Handler: handler, name: name, weight: float64(w)}
+	h.deadline = b.curDeadline + 1/h.weight
+	if h.deadline < b.baseDeadline {
+		h.deadline = math.Ceil((b.baseDeadline-h.deadline)*h.weight) / h.weight
+	}
+	b.curIndex++
```

Comment:
similarly,  I think this write on b.curIndex needs to be locked, no?
also, I don't actually understand why we want to increment curIndex here. Shouldn't it be only updated whenever a handler is selected, just like with curDeadline?

Reply:
index is removed.